openapi3/schema.go

```diff
@@ -1231,3 +1273,113 @@ func isSliceOfUniqueItems(xs []interface{}) bool {
+	if s.Default == nil {
+		return nil
+	}
+	_, ok := s.Default.(string)
+	if !ok {
+		return invalidDefaultValue(s, errors.New("not a string"))
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func validateNumberDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	_, ok := s.Default.(float64)
+	if !ok {
+		return invalidDefaultValue(s, errors.New("not a number"))
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func dfltStr(s *Schema) string { return fmt.Sprintf("%v", s.Default) }
+
+func validateBooleanDefault(s *Schema) (err error) {
+	return validateDefaultProperty(s, func(s *Schema) (err error) { _, err = strconv.ParseBool(dfltStr(s)); return err })
+}
+
+func validateNumberDefault(s *Schema) (err error) {
+	return validateNumberDefaultProperty(s, func(s *Schema) (err error) { _, err = strconv.ParseFloat(dfltStr(s), 64); return err })
+}
+
+func validateFloatDefault(s *Schema) error {
+	return validateNumberDefaultProperty(s, func(s *Schema) (err error) { _, err = strconv.ParseFloat(dfltStr(s), 32); return err })
+}
+
+func validateDoubleDefault(s *Schema) error {
+	return validateNumberDefaultProperty(s, func(s *Schema) (err error) { return nil })
+}
+
+func validateIntegerDefault(s *Schema) error {
+	return validateInt64Default(s)
+}
+
+func validateInt32Default(s *Schema) error {
+	return validateNumberDefaultProperty(s, func(s *Schema) (err error) {
+		f := s.Default.(float64)
+		if f != float64(int32(f)) || math.MinInt32 > f || math.MaxInt32 < f {
+			return errors.New("number not an int or out of range (int32)")
+		}
+		return nil
+	})
+}
+
+func validateInt64Default(s *Schema) error {
+	return validateNumberDefaultProperty(s, func(s *Schema) (err error) {
+		f := s.Default.(float64)
+		if f != float64(int64(f)) || math.MinInt64 > f || math.MaxInt64 < f {
+			return errors.New("number out of range (int64)")
+		}
+		return nil
+	})
+}
+
+func validateStringDefault(s *Schema) error {
+	return validateStringDefaultProperty(s, func(s *Schema) (err error) { return nil })
+}
+
+func validateStringByteDefault(s *Schema) error {
+	return validateStringDefaultProperty(s, func(s *Schema) (err error) {
+		if _, err := base64.StdEncoding.DecodeString(s.Default.(string)); err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+func validateStringDateDefault(s *Schema) error {
+	return validateStringDefaultProperty(s, func(s *Schema) (err error) { _, err = time.Parse("2006-01-02", dfltStr(s)); return err })
+}
+func validateStringDateTimeDefault(s *Schema) error {
+	return validateStringDefaultProperty(s, func(s *Schema) (err error) { _, err = time.Parse(time.RFC3339, dfltStr(s)); return err })
+}
+func validateStringTimeDefault(s *Schema) error {
+	return validateStringDefaultProperty(s, func(s *Schema) (err error) { _, err = time.Parse("00:00:00", dfltStr(s)); return err })
```

Comment:
`03:04:05` is probably what you want here.

Reply:
https://golang.org/pkg/time/#pkg-constants

Reply:
this is a template - any reason for the 345?

Reply:
Yes. Golang uses the numbers to specify what part of the time is being parsed (eg 03 is hours, 04 is minutes, 05 is seconds).  Without that it *might work* but it doesn't follow the time.Parse spec.

Reply:
changed, thanks for the pointer